### ISSUE TITLE
Scenario outline doesn't work inside a Rule scope

### ISSUE THREAD

**Issue by wucke13** [bbqsrc/cucumber-rust#86]
I tried to make an secnario outline - it didn't work. I'm not sure whether this is relevant, but it resides inside a `Rule`. Looking at the matches I seems to just put the placeholder 
`<name of a col in the Examples table>` instead of a value from the regarding table. How is this supposed to work?

This is a minimal example to reproduce the issue:

__Feature__

```
Feature: Guinea Pig
This feature was made to be abused!

  Rule: To them all

    @SomeTag
    Scenario Outline: Parse a Number
      Given the rate of rage is at least <rate of rage> feet per minute
      When the pressure of caviar is most <pressure> miles
      Then tea might help

      Examples:
        | rate of rage | pressure |
        | 1200         | 42       |
        | 2200         | 48       |
        | 3200         | 3        |
```

__Step Definition__

```rust
.given_regex(
    r"the rate of rage is at least (\d+) feet per minute",
    |world, matches, _step| {
        panic!("{}", matches[1]);
        world
    },
);
```

__Output__

```
Feature: Guinea Pig                                                     features/guinea_pig.feature:1:1 

 Rule: To them all                                                      features/guinea_pig.feature:4:3 
  Scenario: Parse a Number                                              features/guinea_pig.feature:8:5 
   - Given the rate of rage is at least <rate of rage> feet per minute  features/guinea_pig.feature:9:7 
     ⚡ Not yet implemented (skipped)
```

Changing the `(\d+)` to `(.+)` reveals:

__Output__

```
Feature: Guinea Pig                                                     features/guinea_pig.feature:1:1 

 Rule: To them all                                                      features/guinea_pig.feature:4:3 
  Scenario: Parse a Number                                              features/guinea_pig.feature:8:5 
   ✘ Given the rate of rage is at least <rate of rage> feet per minute  features/guinea_pig.feature:9:7 
———— [!] Step failed: ————————————————————————————————————————————————————————  test/cucumber.rs:111:17 
  <rate of rage>
———————————————————————————————————————————————————————————————————————————————————————————————————————
```

The placeholder rate of rage is not replaced by an actual number from the table.

**Comment by bbqsrc** [comment]
Closing in favour of https://github.com/bbqsrc/cucumber-rust/issues/64.

**Comment by wucke13** [comment]
The fixes which claim to solve #64 are not making this work, hence I would like to have this issue reopened @bbqsrc

**Comment by wucke13** [comment]
And I can confirm that this issue is caused by the `Rule: ` part. Removing it makes it work with latest master. So @bbqsrc , this indeed is not the same issue a #64 . Unfortunately I can't re open this issue, as you closed it.

**Comment by bbqsrc** [comment]
Right, then it's a slightly different issue. Will reopen. :smile: 

Seemingly, the next obvious step to fix this issue then is to replicate the logic under the rule handling part of the code.

**Comment by tyranron** [comment]
Fixed in #128
